src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -200,8 +214,16 @@ public ActivityLogEntry(AccountAttributes userAccount, boolean isMasquerade, Str
         }
         
         role = changeRoleToAutoIfAutomatedActions(servletName, role);
+        id = googleId + formatCurrentTimeForId();
     }
     
+    private String formatCurrentTimeForId() {
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmssSS");
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
```

Comment:
I think you can use `sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));` here to get the admin time zone instead of using "UTC"